### ISSUE TITLE
all: run tests on windows to make sure things are working

### ISSUE THREAD

**Issue by vangent** [google/go-cloud#1306]

**Comment by vangent** [comment]
@zombiezen have you run a Windows VM locally before to enable tests like this?

**Comment by zombiezen** [comment]
I have run Windows VMs on GCE to check things in the past, but I don't think I've done that for this project.

**Comment by vangent** [comment]
I sent a PR with some fixes for `blob`.

Additionally, `pubsub` tests are failing, I think due to timing issues with the batchers:

```
ok  	gocloud.dev/pubsub	6.980s
2019/03/01 00:03:41 ERROR: martian: failed to round trip: no matching request for &{Method:POST URL:https://sqs.us-east-2.amazonaws.com/ Proto:HTTP/1.1 ProtoMajor:1 ProtoMinor:1 Header:map[User-Agent:[aws-sdk-go/1.16.23 (go1.11.5; windows; amd64)] Content-Type:[application/x-www-form-urlencoded; charset=utf-8] X-Amz-Date:[20190301T000341Z] X-Amz-Security-Token:[FAKE_TOKEN] Accept-Encoding:[gzip] Content-Length:[177] Authorization:[AWS4-HMAC-SHA256 Credential=FAKE_ID/20190301/us-east-2/sqs/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=b99ef74d1510ef7280bcbac4b544580d299d61f6b4f293b75fb88379a698ddca] X-Cloud-Trace-Context:[9af66820f55577c99a1bb504f7e12bcd/1168718018569179706;o=0]] Body:{Reader:0xc0002afef0} GetBody:<nil> ContentLength:177 TransferEncoding:[] Close:false Host:sqs.us-east-2.amazonaws.com Form:map[] PostForm:map[] MultipartForm:<nil> Trailer:map[] RemoteAddr:127.0.0.1:50816 RequestURI:/ TLS:<nil> Cancel:<nil> Response:<nil> ctx:<nil>}
2019/03/01 00:03:49 ERROR: martian: failed to round trip: no matching request for &{Method:POST URL:https://sqs.us-east-2.amazonaws.com/ Proto:HTTP/1.1 ProtoMajor:1 ProtoMinor:1 Header:map[Content-Length:[180] X-Amz-Date:[20190301T000349Z] X-Amz-Security-Token:[FAKE_TOKEN] X-Cloud-Trace-Context:[093d31de147ab86b38d54d6ea6b2c719/12292545737177230626;o=0] User-Agent:[aws-sdk-go/1.16.23 (go1.11.5; windows; amd64)] Authorization:[AWS4-HMAC-SHA256 Credential=FAKE_ID/20190301/us-east-2/sqs/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=b1a813c73e9f58ab822c7ebeb30178c0a05746de16733b2de186ba30e4c05235] Content-Type:[application/x-www-form-urlencoded; charset=utf-8] Accept-Encoding:[gzip]] Body:{Reader:0xc0004171a0} GetBody:<nil> ContentLength:180 TransferEncoding:[] Close:false Host:sqs.us-east-2.amazonaws.com Form:map[] PostForm:map[] MultipartForm:<nil> Trailer:map[] RemoteAddr:127.0.0.1:50828 RequestURI:/ TLS:<nil> Cancel:<nil> Response:<nil> ctx:<nil>}
--- FAIL: TestConformance (23.71s)
    --- FAIL: TestConformance/TestSendReceive (2.88s)
        setup.go:104: Replaying from golden file testdata\TestConformance\TestSendReceive.replay
        drivertest.go:309: pubsub (code=Unknown): SerializationError: failed to decode query XML error response
            	status code: 502, request id: 
            caused by: EOF
    --- FAIL: TestConformance/TestSendReceiveTwo (1.93s)
        setup.go:104: Replaying from golden file testdata\TestConformance\TestSendReceiveTwo.replay
        drivertest.go:309: pubsub (code=Unknown): SerializationError: failed to decode query XML error response
            	status code: 502, request id: 
            caused by: EOF
FAIL
FAIL	gocloud.dev/pubsub/awspubsub	24.604s
ok  	gocloud.dev/pubsub/azurepubsub	1.788s
?   	gocloud.dev/pubsub/driver	[no test files]
?   	gocloud.dev/pubsub/drivertest	[no test files]
--- FAIL: TestConformance (0.18s)
    --- FAIL: TestConformance/TestSendReceive (0.03s)
        replay.go:200: Replaying from golden file testdata\TestConformance\TestSendReceive.replay
        drivertest.go:309: pubsub (code=Unknown): replayer: request not found: subscription:"projects/go-cloud-test-216917/subscriptions/TestConformance_TestSendReceive-subscription-1" max_messages:1 
    --- FAIL: TestConformance/TestSendReceiveTwo (0.02s)
        replay.go:200: Replaying from golden file testdata\TestConformance\TestSendReceiveTwo.replay
        drivertest.go:309: pubsub (code=Unknown): replayer: request not found: subscription:"projects/go-cloud-test-216917/subscriptions/TestConformance_TestSendReceiveTwo-subscription-1" max_messages:1 
FAIL
FAIL	gocloud.dev/pubsub/gcppubsub	1.180s
ok  	gocloud.dev/pubsub/mempubsub	1.966s
```

**Comment by vangent** [comment]
Filed https://github.com/google/go-cloud/issues/1423 for the pubsub stuff.

**Comment by vangent** [comment]
I've confirmed that after the fix for #1525, the Windows build passes.
